charts/cryostat/templates/networkpolicy_ingress.yaml

```diff
@@ -0,0 +1,19 @@
+{{- if ((.Values.networkPolicy.ingress).enabled) }}
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ .Release.Name }}-internal-ingress
+  namespace: {{ .Release.Namespace }}
+spec:
+  podSelector:
+    matchLabels:
+      {{- include "cryostat.selectorLabels" $ | nindent 6 }}
+  ingress:
+    - from:
+      - podSelector:
+          matchLabels:
+            {{- include "cryostat.selectorLabels" $ | nindent 12 }}
+        namespaceSelector:
+          matchLabels:
+            kubernetes.io/metadata.name: {{ .Release.Namespace }}
```

Comment:
> When testing on OpenShift, I seem to be getting 503 errors when trying to access Cryostat through its Route. When the Network Policy is disabled, the route works fine. Not sure why

I wonder if router configurations in different OpenShift clusters play a role here. I tried locally with CRC with OpenShift `4.13` (OpenShift SDN CNI) and OpenShift `v4.15` (OVN-Kubernetes CNI) and was able to access Cryostat via route URL.

If we could add an Ingress rule like below, would it help?

```
  - from:
    - namespaceSelector:
        matchLabels:
          policy-group.network.openshift.io/ingress: ""
```

Reference: https://docs.openshift.com/container-platform/4.16/networking/network_security/network_policy/about-network-policy.html#nw-networkpolicy-allow-from-router_about-network-policy

Reply:
Thanks for finding that @tthvo! That could be what we need. @andrewazores fwiw, this was using a ROSA cluster.

Reply:
Oh nice! Glad that helped ^^